demisto_sdk/commands/common/docker_helper.py

```diff
@@ -467,22 +483,56 @@ def _get_python_version_from_env(env: List[str]) -> Version:
 
 
 @functools.lru_cache
-def get_python_version(image: Optional[str]) -> Version:
-    log_prompt = f"Get python version from image {image}"
-    logger.debug(f"{log_prompt} - Start")
-    if not image or "pwsh" in image or "powershell" in image:
+def get_python_version(
+    image: Optional[str], should_pull_image: bool = True
```

Comment:
I think @GuyAfik meant to give precedence to the dockerhub api instead of the docker client with this argument.
But I don't think we need this argument. Maybe we should use the dockerhub api first when the `DOCKERHUB_USER` and `DOCKERHUB_PASSWORD` is set (to make sure we don't rate limit.

Reply:
@ilaner but i remember you said that its significant to pull the image before we try to query the version directly from the api, mainly used in the pre-commit which contains `DOCKERHUB_USER` and `DOCKERHUB_PASSWORD`.

Reply:
If we get the python version from the API we don't need to pull the image with the docker client, we pull it using the API. That's why we need this variables to exist to avoid pull rate limit when pulling the docker image from the API

Reply:
either way this is logic which should be done from inside the function and it should be consistent. No need to make this an option from outside